src/gauges.rs

```diff
@@ -169,16 +189,78 @@ impl PrometheusGauges {
     }
 
     /// Exports gauges for epoch
-    pub fn export_epoch_info(&self, epoch_info: &EpochInfo) -> anyhow::Result<()> {
-        let first_slot = epoch_info.absolute_slot as i64;
-        let last_slot = first_slot + epoch_info.slots_in_epoch as i64;
+    pub fn export_epoch_info(
+        &self,
+        epoch_info: &EpochInfo,
+        client: &RpcClient,
+    ) -> anyhow::Result<()> {
+        let first_slot = epoch_info.absolute_slot - epoch_info.slot_index;
+        let last_slot = first_slot + epoch_info.slots_in_epoch;
 
         self.transaction_count
             .set(epoch_info.transaction_count.unwrap_or_default() as i64);
         self.slot_height.set(epoch_info.absolute_slot as i64);
         self.current_epoch.set(epoch_info.epoch as i64);
-        self.current_epoch_first_slot.set(first_slot);
-        self.current_epoch_last_slot.set(last_slot);
+        self.current_epoch_first_slot.set(first_slot as i64);
+        self.current_epoch_last_slot.set(last_slot as i64);
+
+        let average_slot_time = (OffsetDateTime::now_utc().unix_timestamp()
+            - client.get_block(first_slot)?.block_time.unwrap())
+            as f64
+            / (epoch_info.slot_index) as f64;
+        self.average_slot_time.set(average_slot_time);
+
+        Ok(())
+    }
+
+    /// Exports information about nodes
+    pub fn export_nodes_info(
+        &self,
+        nodes: &[RpcContactInfo],
+        client: &RpcClient,
+        whitelist: &Whitelist,
+    ) -> anyhow::Result<()> {
+        // Balance of node pubkeys. Only exported if a whitelist is set!
+        if !whitelist.is_empty() {
+            let balances = nodes
+                .iter()
+                .filter(|rpc| {
+                    if whitelist.is_empty() {
+                        true
+                    } else {
+                        whitelist.contains(&rpc.pubkey)
+                    }
+                })
+                .map(|rpc| {
+                    Ok((
+                        rpc.pubkey.clone(),
+                        client.get_balance(&rpc.pubkey.parse()?)?,
+                    ))
+                })
+                .collect::<anyhow::Result<Vec<_>>>()?;
+
+            for (pubkey, balance) in balances {
+                self.node_pubkey_balances
+                    .get_metric_with_label_values(&[&pubkey])
+                    .map(|c| c.set(balance as i64))?;
+            }
+        }
+
+        // Export number of nodes
+        self.nodes.set(nodes.len() as i64);
+
+        // Tally of node versions
+        let versions: HashMap<String, u32> = nodes.iter().fold(HashMap::new(), |mut map, rpc| {
+            *map.entry(rpc.version.clone().unwrap_or_else(|| "unknown".to_string()))
+                .or_insert(0) += 1;
+            map
+        });
+
+        for (version, account) in versions {
```

Comment:
Did you mean `count` rather than `account`?

Reply:
Indeed! Fixed in e099c78a.